src/Behat/Testwork/Argument/PregMatchArgumentOrganiser.php

```diff
@@ -62,7 +62,8 @@ private function cleanupMatchDuplicates(array $match)
         $arguments = array();
 
         $keys = array_keys($cleanMatch);
-        for ($keyIndex = 0; $keyIndex < count($keys); $keyIndex++) {
+        $keysCount = count($keys);
+        for ($keyIndex = 0; $keyIndex < $keysCount; $keyIndex++) {
```

Comment:
How about `foreach()`?

Reply:
they keyIndex is sometimes incremented manually inside the loop, so a foreach cannot be used.

Reply:
`for ($keyIndex = 0, $keysCount = count($keys); ...)` would be more semantically correct IMO